### PR TITLE
explain payloads

### LINKED ISSUE #2169: Payload is undefined jargon
The term 'payload' is Juju specific jargon and never defined, making the documentation for the payload-* hook environment tools unintelligible. It is certainly not a standard definition of a payload, having something to do with docker or virtual machines per the examples. I suspect this is why I've never seen the feature used.

### PR DESCRIPTION
**Pull request by pmatulis**
FWIW
resolves #2169 
applies to 1.25 and 2.x

### CODE REVIEW

File: src/en/authors-charm-metadata.md [hunk 1]
```diff
@@ -125,23 +125,33 @@ resources:
 
 ## Payloads
 
-You can use the `payloads` section of metadata.yaml to help the user of a charm
-better understand the purpose of payloads such as LXC, KVM and docker. This is
-especially useful in large and complex deployments.
+Payloads provide a means for the charm author to get information out of the
```
Comment (evilnick) [inline src/en/authors-charm-metadata.md]: /out of/from/ ?

File: src/en/authors-charm-metadata.md [hunk 2]
```diff
@@ -125,23 +125,33 @@ resources:
...
+services deployed by a charm. This is especially useful in large and complex
```
Comment (evilnick) [inline src/en/authors-charm-metadata.md]: someone is going to complain about the use of 'services'.  Perhaps we can avoid and simplify by just saying "... information from a deployed charm"

File: src/en/authors-charm-metadata.md [hunk 3]
```diff
@@ -125,23 +125,33 @@ resources:
...
+deployments. For instance, the author may want to check the status of some
+element of the services such as a Docker container.
```
Comment (evilnick) [inline src/en/authors-charm-metadata.md]: of the deployment ?

File: src/en/authors-charm-metadata.md [hunk 4]
```diff
@@ -125,23 +125,33 @@ resources:
...
-Payloads are defined by creating a class for the payload, such as `monitoring`
-or `kvm- guest`, and then assigning a type:
+Payloads are defined in the `payloads` section of `metadata.yaml` by assigning
+a class and type. A class defines the name of the payload and the type
+describes the nature of the payload. Both are author-defined and are not
+validated by Juju.
+
+The most common types of payload are based on Docker, KVM, and LXD.
+
+As an example, below the following class/type pairs are defined:
```
Comment (evilnick) [inline src/en/authors-charm-metadata.md]: for some reason I feel strongly there should be an additional ',' after below

File: src/en/authors-charm-metadata.md (hunk 1 above)
Comment (pmatulis) [reply src/en/authors-charm-metadata.md]: OK

File: src/en/authors-charm-metadata.md (hunk 2 above)
Comment (pmatulis) [reply src/en/authors-charm-metadata.md]: OK

File: src/en/authors-charm-metadata.md (hunk 3 above)
Comment (pmatulis) [reply src/en/authors-charm-metadata.md]: OK

File: src/en/authors-charm-metadata.md (hunk 4 above)
Comment (pmatulis) [reply src/en/authors-charm-metadata.md]: OK

**Pull request merged**